### PR TITLE
Build tree from values instead of wrapping and unwrapping

### PR DESCRIPTION
**Pull request by HuwCampbell**
This is just a minor cleanup, there should be no function differences.

As I mentioned earlier, my original version wrapped and unwrapped trees a bit much, making the code a bit harder to follow.

Pulling out a helper for `integral_` means we can build on values only.

### CODE REVIEW

File: hedgehog/src/Hedgehog/Internal/Prelude.hs [hunk 1]
```diff
@@ -8,10 +8,15 @@ module Hedgehog.Internal.Prelude (
...
 
 import           Control.Monad.Fail (MonadFail)
 
 import           Data.Semigroup (Semigroup(..))
 
 import           Prelude hiding (filter, print, map)
+
+(...) :: (c -> d) -> (a -> b -> c) -> a -> b -> d
+(...) = (.) . (.)
+{-# INLINE (...) #-}
```
Comment (jacobstanley) [inline hedgehog/src/Hedgehog/Internal/Prelude.hs:22]: I think I'd rather not have this, this module was more intended for compatibility than introducing custom operators. I'd like to keep things as vanilla as possible for now.

Maybe you can convince me, but it shouldn't be part of this PR.

Comment (moodmosaic) [reply hedgehog/src/Hedgehog/Internal/Prelude.hs:22]: @HuwCampbell, perhaps also we can move this to the let binding above where it’s used?

This clashes with ES6 spread syntax, so perhaps we could just rewrite it without (...) altogether.

**Review by moodmosaic — COMMENTED** [review]
Great work! See @jacobstanley's comment and the one I also left. I'd be nice to have this PR on this release.

File: hedgehog/src/Hedgehog/Internal/Prelude.hs (hunk 1 above)
Comment (HuwCampbell) [reply hedgehog/src/Hedgehog/Internal/Prelude.hs]: Gone.  I've just eta expanded both arguments.

I do like the owl (called so as its implementation is `(.) . (.)`) as it's just another simple form of function composition, but I understand what you're going for with this module.

**Review by moodmosaic — APPROVED** [review]
👍

**Review by jacobstanley — APPROVED** [review]
